persist-cli/src/main/java/io/ballerina/persist/utils/BalProjectUtils.java

```diff
@@ -85,4 +95,69 @@ private static String fotmatError(String errorMessage, Path path) {
         }
         return errorMessage;
     }
+
+    public static EntityMetaData readBalFiles(String sourcePath) throws BalException {
+        ArrayList<Entity> returnMetaData = new ArrayList<>();
+        ArrayList<ModuleMemberDeclarationNode> returnModuleMembers = new ArrayList<>();
+        Path dirPath = Paths.get(sourcePath);
+        List<Path> fileList;
+
+        try (Stream<Path> walk = Files.walk(dirPath)) {
+            boolean skipValidation = false;
```

Comment:
Lets simplify the logic here,
1. Get directories
2. Skip `client` sub module
3. Process the rest

The syntax diagnostic error check can be moved out of the function.

Reply:
Refactored in [27f56f0](https://github.com/ballerina-platform/persist-tools/pull/60/commits/27f56f00a2995030037b8512dfb7e9848701706a). 

@sahanHe please review this changes